client/components/date-range/test/index.js

```diff
@@ -0,0 +1,632 @@
+
+			expect( actual ).toEqual( expected );
+		} );
+
+		test( 'should disable dates after lastSelectableDate when set', () => {
+			const today = new Date();
+			const wrapper = shallow( <DateRange moment={ moment } lastSelectableDate={ today } /> );
+			const datePicker = wrapper.find( DatePicker );
+
+			const expected = [
+				{
+					after: today,
+				},
+			];
+
+			const actual = datePicker.props().disabledDays;
+
+			expect( actual ).toEqual( expected );
+		} );
+
+		test( 'should disable DatePicker UI for months previous to firstSelectableDate when set', () => {
+			const today = new Date();
+
+			const wrapper = shallow( <DateRange moment={ moment } firstSelectableDate={ today } /> );
+			const datePicker = wrapper.find( DatePicker );
+
+			const expected = today;
+			const actual = datePicker.props().fromMonth;
+
+			expect( actual ).toEqual( expected );
+		} );
+
+		test( 'should disable DatePicker UI for months after lastSelectableDate when set', () => {
+			const today = new Date();
+
+			const wrapper = shallow( <DateRange moment={ moment } lastSelectableDate={ today } /> );
+			const datePicker = wrapper.find( DatePicker );
+
+			const expected = today;
+			const actual = datePicker.props().toMonth;
+
+			expect( actual ).toEqual( expected );
+		} );
+	} );
+
+	describe( 'Input elements', () => {
+		let startDate;
+		let endDate;
+		let momentStartDate;
+		let momentEndDate;
+
+		beforeEach( () => {
+			startDate = '2018-04-20';
+			endDate = '2018-05-28';
+			momentStartDate = moment( startDate );
+			momentEndDate = moment( endDate );
+		} );
+
+		test( 'should see inputs reflect date picker selection', () => {
+			const wrapper = shallow( <DateRange moment={ moment } /> );
+
+			const expectedStart = '2018-04-03';
+			const expectedEnd = '2018-04-29';
+
+			const newStartDate = moment( expectedStart );
+			const newEndDate = moment( expectedEnd );
+
+			// Select dates using API
+			wrapper.instance().onSelectDate( newStartDate );
+			wrapper.instance().onSelectDate( newEndDate );
+
+			// Force update
+			wrapper.update();
+
+			const dateRangeInputs = wrapper.find( DateRangeInputs );
+
+			expect( dateRangeInputs.props() ).toEqual(
+				expect.objectContaining( {
+					startDateValue: dateToLocalString( newStartDate ),
+					endDateValue: dateToLocalString( newEndDate ),
+				} )
+			);
+		} );
+
+		test( 'should update start date selection on start date input blur event', () => {
+			const wrapper = shallow( <DateRange moment={ moment } /> );
+
+			wrapper.instance().handleInputBlur( '04/20/2018', 'Start' );
+
+			expect( wrapper.state().startDate.format( 'L' ) ).toEqual( momentStartDate.format( 'L' ) );
+		} );
+
+		test( 'should update end date selection on end date input blur event', () => {
+			const wrapper = shallow( <DateRange moment={ moment } /> );
+
+			wrapper.instance().handleInputBlur( '05/28/2018', 'End' );
+
+			expect( dateToLocalString( wrapper.state().endDate ) ).toEqual(
```

Comment:
What's the reason for the `endDate` test to be different to the `startDate` test in terms of comparing the formatted values?

Reply:
Human error. Fixed.